### PR TITLE
Add preConfigureCallback check for GCP creds

### LINKED ISSUE #860: improve gcp config error message?
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

was trying to deploy a GKE cluster and I thought the error message could be better around providing a command to configure
- project
- zone

[asciicast](https://asciinema.org/a/bM0L9E1c8DSO7qKezySQuWxvt)

cc @guineveresaenger

### PR DESCRIPTION
**Pull request by guineveresaenger**
Add authentication error messages

- sketch out callback func
- use given library to validate
- Error early if Project not set. Catch default app error.
- finalize messaging
- make sdks with go 1.19

Fixes #860

### CODE REVIEW

**Comment by guineveresaenger** [comment]
Converting to draft because this check ignores the Project having been set via `gcloud config set project`.

Suggestions welcome.

File: provider/resources.go [hunk 1]
```diff
@@ -156,6 +159,65 @@ func boolRef(b bool) *bool {
...
+		return fmt.Errorf("unable to find required configuration setting: GCP Project\n" +
+			"Set the GCP Project by using:\n" +
+			"\t`pulumi config set gcp:project <project>`")
+	}
+
+	// check if skipCredentialsValidation is True, if yes, exit early
+	if val, ok := vars["skipCredentialsValidation"]; ok {
+		if val.IsBool() {
+			return nil
+		}
```
Comment (lblackstone) [inline provider/resources.go:194]: This doesn't look right to me. I think this will return true if any bool is set.

File: provider/resources.go [hunk 2]
```diff
@@ -156,6 +159,65 @@ func boolRef(b bool) *bool {
...
+	}
+	return ""
+}
+
+func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) error {
+
+	// explicitly check to make sure that the user has a project available before we do
+	// anything with the provider
+	project := stringValue(vars, "project", []string{"GOOGLE_PROJECT", "GOOGLE_CLOUD_PROJECT",
```
Comment (lblackstone) [inline provider/resources.go:181]: I'd suggest keeping each env var on a separate line to improve readability here.

File: provider/resources.go [hunk 3]
```diff
@@ -3,14 +3,17 @@
...
 	"fmt"
+	"os"
 	"path/filepath"
 	"strings"
 	"unicode"
 
 	google "github.com/hashicorp/terraform-provider-google-beta/google-beta"
 	"github.com/pulumi/pulumi-gcp/provider/v6/pkg/version"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
```
Comment (lblackstone) [inline provider/resources.go:16]: I just wanted to make sure that it's intended to use both v1 and v2 versions of `tfshim`?

**Review by lblackstone — COMMENTED** [review]
I like the overall direction here, but had a couple questions.

File: provider/resources.go (hunk 1 above)
Comment (guineveresaenger) [reply provider/resources.go:194]: It turns out to be unnecessary in the first place -thank you for catching!

File: provider/resources.go (hunk 3 above)
Comment (guineveresaenger) [reply provider/resources.go:16]: yeah - the preConfigureCallback takes a shim :/

**Review by lblackstone — APPROVED** [review]

**Review by stack72 — APPROVED** [review]

**Pull request merged**